### PR TITLE
#191 - Added date parsing to ChangelogEntry

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #191 
Added `date` method to `ChangelogEntry` for parsing date from source string

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#237](https://github.com/artipie/rpm-adapter/issues/237) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/ddeff6ae-22b3-40a4-919c-5af3c126b885); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olegmoz** [comment]
@paulodamaso ping

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) I see that you unassigned this issue; the order is still assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) though; to cancel the order use `refuse`, as in [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#237](https://github.com/artipie/rpm-adapter/issues/237) is assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/8da3b8a2-b126-428e-a3b1-f031e5b4942e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/rpm/meta/ChangelogEntry.java [hunk 1]
```diff
@@ -64,6 +71,25 @@ String author() {
         return this.matcher().group("author");
     }
 
+    /**
+     * Read date.
+     *
+     * @return Date in UNIX time.
+     */
+    int date() {
+        final Matcher matcher = this.matcher();
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/meta/ChangelogEntry.java:80]: @olegmoz please inline this variable

File: src/main/java/com/artipie/rpm/meta/ChangelogEntry.java [hunk 2]
```diff
@@ -64,6 +71,25 @@ String author() {
...
+        final String str = matcher.group("date");
+        final SimpleDateFormat format = new SimpleDateFormat("EEE MMM dd yyyy", Locale.US);
+        format.setTimeZone(TimeZone.getTimeZone("UTC"));
+        final Date date;
+        try {
+            date = format.parse(str);
+        } catch (final ParseException ex) {
+            throw new IllegalStateException(String.format("Failed to parse date: '%s'", str), ex);
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/rpm/meta/ChangelogEntry.java:88]: @olegmoz let's add test method for this branch

**Comment by 0crat** [comment]
Job was finished in 27 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)[/z](https://www.0crat.com/u/olenagerasimova) didn't make CR comments

**Pull request merged**